Assignment 2/test/Vault.t.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "src/Vault.sol";
+import "src/Coll.sol";
+import "yield-utils-v2/token/IERC20.sol";
+
+/// @title Test state with just an initialized `Vault` contract.
+/// @notice This test state sets up a `Vault` with `Coll` as token.
+/// @dev The setup is straightforward and happens in `setUp`.
+/// @author Thomas den Hollander
+abstract contract ZeroState {
+    Vault vault;
+    Coll token;
+
+    event Deposit(address indexed account, uint256 deposited);
+    event Withdrawal(address indexed account, uint256 withdrawn);
+
+    function setUp() public virtual {
+        token = new Coll();
+        vault = new Vault(token);
+    }
+}
+
+/// @notice Test state where the contract has minted some tokens.
+abstract contract MintedTokenState is ZeroState {
+    /// @notice The amount of tokens that the contract should own at the start
+    ///     of tests.
+    uint256 constant tokensMinted = 1_000_000_000_000_000_000;
+
+    function setUp() public virtual override {
+        super.setUp();
+        token.mint(address(this), tokensMinted);
```

Comment:
ur using this test contract as the user.  a more clear pattern is to setup a user address in state and refer to it.  everytime u want that user to do something, use vm.prank.   see [calnix example](https://github.com/calnix/Basic-Vault/pull/2/files#diff-63defda6a33f4b1b264b918d2690459a930533f21cd6f513259ea2b7cb632f5bR23-R24)

it avoids confusion, and becomes more useful when u have more than one user